playlistmanager.lua

```diff
@@ -1071,63 +1074,113 @@ end
 mp.observe_property('playlist-count', "number", function()
   if playlist_visible then showplaylist() end
   if settings.prefer_titles == 'none' then return end
+  if not settings.resolve_titles and not settings.resolve_local_titles then return end
   -- resolve titles
   resolve_titles()
 end)
 
---resolves url titles by calling youtube-dl
+--resolves titles by calling youtube-dl or ffprobe
 function resolve_titles()
-  if not settings.resolve_titles then return end
   local length = mp.get_property_number('playlist-count', 0)
   if length < 2 then return end
   local i=0
   -- loop all items in playlist because we can't predict how it has changed
   while i < length do
     local filename = mp.get_property('playlist/'..i..'/filename')
     local title = mp.get_property('playlist/'..i..'/title')
-    if i ~= pos
-      and filename
-      and filename:match('^https?://')
-      and not title
-      and not url_table[filename]
-      and not requested_urls[filename]
-    then
-      requested_urls[filename] = true
-
-      local args = { settings.youtube_dl_executable, '--no-playlist', '--flat-playlist', '-sJ', filename }
-      local req = mp.command_native_async(
-        {
-          name = "subprocess",
-          args = args,
-          playback_only = false,
-          capture_stdout = true
-        }, function (success, res)
-            if res.killed_by_us then
-              msg.verbose('Request to resolve url title ' .. filename .. ' timed out')
+
+    local function resolve_url()
```

Comment:
Maybe this could be a top level function and take filename as parameter. Same for resolve_local function.